### PR TITLE
refactor(icon): update icon component

### PR DESCRIPTION
**Pull request by lauralucca**
# Description

This updates the Icon component to better fit our code and UX standards, as described in [DSY-1965 Component | Icon [RN]](https://natura.atlassian.net/browse/DSY-1965). The changes included are:

- [x] Adding `accessibilityRole` as a property you can define when using the component;
- [x] Changing the default `accessibilityRole` from "text" to "image";
- [x] Fixing the `Icon` size options that weren't working in the `Interactive` section of the sample app;
- [x] Refactoring and adding new tests to `Icon` component

No dependencies were added or removed for this change.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -1,77 +1,43 @@
...
+  name = 'outlined-default-mockup',
   testID = `icon-${name}`,
   theme,
   size = 'standard',
 }: IconProps) => {
-  const unicodeName = icons[name]
-    ? icons[name].replace('%', '\\')
-    : icons[defaultIconName];
-
+  const unicodeName = icons[name].replace('%', '\\');
```
Comment (arielwb) [inline src/components/Icon/Icon.tsx:31]: voltar com o ternario que estava aqui.
se eu passar um  nome invalido vai estourar um erro no console

File: src/components/Icon/Icon.test.tsx [hunk 2]
```diff
@@ -23,18 +22,26 @@ const sampleNames: Array<IconName> = [
 ];
 
 describe('Icon component', () => {
```
Comment (arielwb) [inline src/components/Icon/Icon.test.tsx:24]: faltou testar a logica de getIconColor

File: src/components/Icon/Icon.tsx (hunk 1 above)
Comment (arielwb) [reply src/components/Icon/Icon.tsx:31]: e testar isso tbm

File: src/components/Icon/Icon.test.tsx [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
-    <Icon {...props} />
-  </ThemeProvider>,
-));
+import { Icon } from './Icon';
+import { renderWithTheme } from '../../../test/testHelpers';
+import { IconProps } from './Icon.types';
+
+jest.mock('../../common/themeSelectors', () => (
+  {
+    getColorByName: () => '#BBBBBB',
```
Comment (arielwb) [inline src/components/Icon/Icon.test.tsx:9]: testar se isso foi chamado

File: src/components/Icon/Icon.stories.tsx [hunk 4]
```diff
@@ -1,29 +1,80 @@
...
+  <StoryContainer title="Sizes">
+    <Icon size="small" />
+    <Icon size="medium" />
+    <Icon size="large" />
+    <Icon size="huge" />
+  </StoryContainer>
 );
 
 /* eslint-disable sort-keys */
 const iconColors = {
```
Comment (yasminnvaz) [inline src/components/Icon/Icon.stories.tsx:77]: @arielwb você enxerga alguma forma de fazer isso melhor? Tanto para tamanhos quanto cores?

File: ios/NaturaReactComponents/main.m [hunk 5]
```diff
@@ -11,6 +11,6 @@
 
 int main(int argc, char * argv[]) {
   @autoreleasepool {
-    return UIApplicationMain(argc, argv, nil, NSStringFromClass([AppDelegate class]));
+     return UIApplicationMain(argc, argv, nil, NSStringFromClass([AppDelegate class]));
```
Comment (lauralucca) [inline ios/NaturaReactComponents/main.m:14]: eu não sei como isso aconteceu. faz sentido tentar tirar @yasminnvaz ?

Comment (yasminnvaz) [reply ios/NaturaReactComponents/main.m:14]: pelo visto foi apenas formatação, não vejo problemas em manter 🤔

**Pull request merged**